ksqldb-parser/src/main/java/io/confluent/ksql/parser/KsqlParserUtil.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.parser;
+
+import com.google.common.collect.ImmutableSet;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.Vocabulary;
+
+public final class KsqlParserUtil {
+
+  private KsqlParserUtil() {
+  }
+
+  public static boolean isReserved(final String token) {
+
+    final SqlBaseLexer sqlBaseLexer = new SqlBaseLexer(
+            new CaseInsensitiveStream(CharStreams.fromString(token)));
+    final CommonTokenStream tokenStream = new CommonTokenStream(sqlBaseLexer);
+    final SqlBaseParser sqlBaseParser = new SqlBaseParser(tokenStream);
+
+    final SqlBaseParser.NonReservedContext nonReservedContext = sqlBaseParser.nonReserved();
```

Comment:
I would spend some time with a debugger and see if there's another way we can check if this doesn't match other than if there's an exception. There might be a cleaner way, I just hacked this together really quickly to prove that it was possible. Let me know if you can't find anything

Reply:
This seems to be the only way to do it for me. Tried to condense it down (seems to be a lot of work for checking something simple)